lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -21,6 +23,23 @@
     'lti',
 ]
 
+# exclude these in Metrics
+non_student_roles = ['instructor', 'staff']
+
+def get_non_student_list(course_key):
+    # find all user_ids with instructor or staff roles in student_courseaccessrole table
+    non_student_idlist=[]
+    non_students = CourseAccessRole.objects.select_related('user').filter(
+        course_id=course_key,
+        role__in=non_student_roles,
+    ).values('user__id')
+
+    for non_student in non_students:
+        curr_user_id = non_student['user__id']
+        non_student_idlist.append(curr_user_id);
```

Comment:
ah.  list comprehensions are a really nice feature of python.  The more "pythonic" way to write this would have been to write
```
non_student_idlist = [non_student['user__id'] for non_student in non_students]
```
(and you wouldn't have to initialize `non_student` before)